js/liqui.js

```diff
@@ -390,6 +399,8 @@ module.exports = class liqui extends Exchange {
             if (this.last_json_response) {
                 let message = this.safeString (this.last_json_response, 'error');
                 if (message) {
+                    // FIXME: since error handling is in handleErrors() now this will never get executed
```

Comment:
For some (maybe valid) reason cancellation error was handled here, not in `request()`.

If the reason was valid this chunk of code needs to be reconsidered.
If it's not it maybe safely deleted.

Reply:
They will return a HTTP error code on an already-canceled order. I would suggest adding that to the test case, can you do that, please? That would uncover a few other exchanges, I believe.

Reply:
... we may end up removing this section then.

Reply:
Also, they will often return "not available" or "external service unavailable" on a canceled order making it much more difficult to detect properly.

Reply:
> They will return a HTTP error code on an already-canceled order. I would suggest adding that to the test case, can you do that, please?

Ok.

> Also, they will often return "not available" or "external service unavailable"

I would leave that for users' to decide. I expect them to retry if exchange is not available (that's what I do).

Reply:
> They will return a HTTP error code on an already-canceled order. I would suggest adding that to the test case, can you do that, please?

Actually, I remember I did it but then switched it off because some exchanges reported 'ok' while others reported with `OrderNotFound`. Will revert it back anyway.

Reply:
Ok, I put the test back, liqui reports `OrderNotFound`.

If you're ok with my proposal to offload handling of "not availble" to users, I'll remove this piece.

Reply:
> I'll remove this piece

How about moving it into handleErrors? ) I think we should still keep that "not found" code, when Liqui is healthy enough to reply properly. )

Reply:
It's there, that's why this chunk will never be reached from now on anyway.

Reply:
Ah, ok, skip the last comment.

Reply:
Are you sure? "not found" != "not available", these two are different codes ) Or am I missing something?

Reply:
Ah, ok, I noticed your comment too. Looks like we misunderstood each other.

```
:air:~/src/ccxt max$ node examples/js/cli.js liqui createLimitBuyOrder ETH/USDT 1 1
{ id:   "203572644",
  timestamp:    1516027268321,
   datetime:   "2018-01-15T14:41:08.321Z",
     status:   "open",
     symbol:   "ETH/USDT",
       type:   "limit",
       side:   "buy",
      price:    1,
       cost:    0,
     amount:    1,
  remaining:    1,
     filled:    0,
        fee:    undefined }
:air:~/src/ccxt max$ node examples/js/cli.js liqui cancelOrder 203572644
{ success:    1,
   return: { order_id:    203572644,
                funds: {  eth: 0,
                         usdt: 1  } },
     stat: {  isSuccess:  true,
             serverTime: "00:00:00.0091973",
                   time: "00:00:00.0477042",
                 errors:  null               }         }
:air:~/src/ccxt max$ node examples/js/cli.js liqui cancelOrder 203572644
[OrderNotFound] liqui cancelOrder() error: {"success":0,"code":833,"error":"Order with id 203572644 was not found."}
```

I thought that you wanted to catch subsequent "exchange not available" errors while my proposal was to let users retry.

So let's be clear now: `OrderNotFound` works as expected. How about those "exchange not available" errors? It might be that the exchange **indeed** is not available, I wouldn't return `OrderNotFound`, it might be a false positive.

Reply:
Ok, this is the last issue that remains unresolved.

Now canceling already canceled order results in `OrderNotFount` exception. Let me know if you want some extra behavior above that.